app/models/user.rb

```diff
@@ -14,4 +19,8 @@ def recalculate_order
   def not_in_queue?(video)
     !queue_items.map(&:video_id).include?(video.id)
   end
+
+  def can_follow?(another_user)
+    !(self.friends.include?(another_user) || self == another_user)
```

Comment:
There is no need to use `self.friends`, `friends` would work fine and is the recommended Ruby style.